libraries/helpers.rb

```diff
@@ -27,8 +25,40 @@ def dhparam_pem_valid?(dhparam_pem_path)
       dhparam.params_ok?
     end
 
+    def key_file_valid?(key_file_path, key_password = nil)
+      # Check if the key file exists
+      # Verify the key file contains a private key
+      return false unless File.exist?(key_file_path)
+      key = OpenSSL::PKey::RSA.new File.read(key_file_path), key_password
+      key.private?
+    end
+
+    # Generators
     def gen_dhparam(key_length)
+      fail ArgumentError, 'Key length must be a power of 2 greater than or equal to 1024'\
+      unless key_length_valid?(key_length)
+
       OpenSSL::PKey::DH.new(key_length, 2)
     end
+
+    def gen_rsa_key(key_length)
+      fail ArgumentError, 'Key length must be a power of 2 greater than or equal to 1024'\
+      unless key_length_valid?(key_length)
+
+      OpenSSL::PKey::RSA.new(key_length)
+    end
+
+    # Key manipulation helpers
+    # Returns a pem string
+    def encrypt_rsa_key(rsa_key, key_password)
+      fail TypeError, 'rsa_key must be a Ruby OpenSSL::PKey::RSA object'\
```

Comment:
Minor style point, but maybe just change these into 

```ruby
unless COND
   CODE
end
```

We tend to only use the \ to continue the next line for strings.